src/api/search_index_manager.py

```diff
@@ -0,0 +1,340 @@
+        :raises: Value error if both dimensions of embedding model and vector_index_dimensions are not set
+                 or both of them set and they do not equal each other.
+        """
+        if vector_index_dimensions is None:
+            if self._dimensions is None:
+                raise ValueError(
+                    "No embedding dimensions were provided in neither dimensions in the constructor nor in vector_index_dimensions"
+                    "Dimensions are needed to build the search index, please provide the vector_index_dimensions.")
+            vector_index_dimensions = self._dimensions
+        if self._dimensions is not None and vector_index_dimensions != self._dimensions:
+            raise ValueError("vector_index_dimensions is different from dimensions provided to constructor.")
+        return vector_index_dimensions
+
+    async def ensure_index_created(self, vector_index_dimensions: Optional[int] = None) -> None:
+        """
+        Get the search index. Create the index if it does not exist.
+
+        :param vector_index_dimensions: The number of dimensions in the vector index. This parameter is
+               needed if the embedding parameter cannot be set for the given model. It can be
+               figured out by loading the embeddings file, generated by build_embeddings_file,
+               loading the contents of the first row and 'embedding' column as a JSON and calculating
+               the length of the list obtained.
+               Also please see the embedding model documentation
+               https://platform.openai.com/docs/models#embeddings
+        :raises: Value error if both dimensions of embedding model and vector_index_dimensions are not set
+                 or both of them set and they do not equal each other.
+        """
+        vector_index_dimensions = self._check_dimensions(vector_index_dimensions)
+        if self._index is None:
+            self._index = await SearchIndexManager.get_or_create_search_index(
+                self._endpoint,
+                self._credential,
+                self._index_name,
+                vector_index_dimensions)
+
+    @staticmethod
+    async def index_exists(
+        endpoint: str,
+        credential: AsyncTokenCredential,
+        index_name: str) -> bool:
+        """
+        Check if index exists.
+
+        :param endpoint: The search end point to be used.
+        :param credential: The credential to be used for the search.
+        :param index_name: The name of an index to get or to create.
+        :return: True if index already exists.
+        """
+        exists = False
+        async with SearchIndexClient(endpoint=endpoint, credential=credential) as ix_client:
+            try:
+                await ix_client.get_index(index_name)
+                exists = True
+            except ResourceNotFoundError:
+                pass
+        return exists
+
+    @staticmethod
+    async def get_or_create_search_index(

```

Comment:
nit. compared to create_index and other xxx_index methods do not have search

Could this be then get_or_create_index ?

Reply:
Fixed.